kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemDataset.java

```diff
@@ -463,6 +469,13 @@ public long getLastModified() {
         throw new DatasetIOException("Cannot find last modified time of of " + dir, e);
       }
     }
+
+    // use the ready time if it is more recent
+    long viewModified = unbounded.getLastModified();
```

Comment:
I think I followed, but just to be sure, I fully delegated this method to the "unbounded.getLastModified" since at this point it would be duplicating the above filesystem traversal.